src/map/unit.hpp

```diff
@@ -83,21 +83,12 @@ struct unit_data {
 };
 
 struct view_data {
-	uint16 class_;
-	t_itemid
-		weapon,
-		shield, //Or left-hand weapon.
-		robe,
-		head_top,
-		head_mid,
-		head_bottom;
-	uint16
-		hair_style,
-		hair_color,
-		cloth_color,
-		body_style;
+	int32 look[LOOK_MAX];
 	char sex;
 	unsigned dead_sit : 2; // 0: Standing, 1: Dead, 2: Sitting
+
+public:
+	void update( map_session_data& sd, _look look );
```

Comment:
this should be a method in map_session_data, then you could just do `sd.update(look)`

Reply:
It should be like this, but actually it should accept a block_list as parameter later on.
I just left it like this now, as right now only the player specific behavior exists in here.
Other cases than just weapon and shield should probably also be merged into this function later on.
Since the view_data objects dont have a link to their "master" - only in the other direction (so from block_list, map_session_data, etc.) updates are possible.
That is why for me it is a function in the view_data class/struct and not a helper/convenience function in the "master" class.

Or do you have a better idea?

Reply:
if theres anything that's common across all view_datas, then we can put that in view_data, but for player specific, i think we should put it in `map_session_data::updateLook(_look look)`, and then we can call `vd.update(look)` in it

Reply:
done.